src/tests/ftest/launch.py

```diff
@@ -659,233 +659,163 @@ def get_test_files(test_list, args, yaml_dir, vmd_flag=False):
     return test_files
 
 
-def get_nvme_replacement(args):
+def get_device_replacement(args):
     """Determine the value to use for the '--nvme' command line argument.
 
-    Parse the lspci output for any NMVe devices, e.g.
-        $ lspci | grep 'Non-Volatile memory controller:'
-        5e:00.0 Non-Volatile memory controller:
-            Intel Corporation NVMe Datacenter SSD [3DNAND, Beta Rock Controller]
-        5f:00.0 Non-Volatile memory controller:
-            Intel Corporation NVMe Datacenter SSD [3DNAND, Beta Rock Controller]
-        81:00.0 Non-Volatile memory controller:
-            Intel Corporation NVMe Datacenter SSD [Optane]
-        da:00.0 Non-Volatile memory controller:
-            Intel Corporation NVMe Datacenter SSD [Optane]
-
-    Optionally filter the above output even further with a specified search
-    string (e.g. '--nvme=auto:Optane'):
-        $ lspci | grep 'Non-Volatile memory controller:' | grep 'Optane'
-        81:00.0 Non-Volatile memory controller:
-            Intel Corporation NVMe Datacenter SSD [Optane]
-        da:00.0 Non-Volatile memory controller:
-            Intel Corporation NVMe Datacenter SSD [Optane]
+    Determine if the specified hosts have homogeneous NVMe drives (either standalone or VMD
+    controlled) and use these values to replace placeholder devices in avocado test yaml files.
+
+    Supported auto '--nvme' arguments:
+        auto[:filter]       = select any PCI domain number of a NVMe device or VMD controller
```

Comment:
Is there a chance that you could end up with a mix of NVMe and VMD, probably a result of misconfiguration, that may give undefined results?

Reply:
It's not a blocker for this PR by any means but just something to consider going forward. We probably will need some more validation of number of backing devices across engines for VMD configurations and warnings against having mixed VMD/non-VMD configurations.

Reply:
Yes if the cluster was configured with both VMD and non-VMD NVMe we could end up with a list of both provided in the server configuration file.  For our functional tests to get into this situation it would require there to be less that two VMDs auto-detected as most (if not all) of our functional tests that specify a `bdev_list` only include two device placeholders.  It does sound like we should  have some warnings for mixed VMD/non-VMD configs.

Reply:
When running manually if the user is expecting to use only one type then `--nvme=auto_vmd` or `--nvme=auto_nvme` can be used to limit which addresses are used to substitute the values in each test yaml file.